packages/docker-engine-29/0004-force-the-containerd-snapshotter-to-be-used.patch

```diff
@@ -0,0 +1,31 @@
+From 47adb0c8cc87fa77030df05051f323c28e0a16ca Mon Sep 17 00:00:00 2001
+From: Ben Cressey <bcressey@amazon.com>
+Date: Sat, 8 Nov 2025 06:35:31 +0000
+Subject: [PATCH] force the containerd snapshotter to be used
+
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ daemon/daemon.go | 9 ++++++---
+ 1 file changed, 6 insertions(+), 3 deletions(-)
+
+diff --git a/daemon/daemon.go b/daemon/daemon.go
+index ce06f8e952..8126bf69aa 100644
+--- a/daemon/daemon.go
++++ b/daemon/daemon.go
+@@ -1145,10 +1145,13 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
+ 	}
+ 
+ 	var migrationConfig migration.Config
+-	tryGraphDriver, err := isGraphDriver(driverName)
+-	if err != nil {
+-		return nil, err
++	// Force containerd snapshotter to always be used
++	_ = isGraphDriver
++	// Map graphdriver names to snapshotter names
++	if driverName == "overlay2" {
++		driverName = "overlayfs"
+ 	}
++	tryGraphDriver := false
```

Comment:
It's fine to take this since it works, but doing a deeper analysis of why we end up using `graphdriver` without this patch would be great.

It could be that there's a simple tweak to `daemon.json` that stops it from diverting onto the legacy path.